datumaro/components/dataset.py

```diff
@@ -906,8 +963,18 @@ def import_from(cls, path: str, format: Optional[str] = None,
         return dataset
 
     @staticmethod
-    def detect(path: str, env: Optional[Environment] = None,
-            depth: Optional[int] = 1) -> str:
+    def detect(path: str, *,
+            env: Optional[Environment] = None, depth: int = 1) -> str:
+        """
+        Attempts to detect dataset format of a given directory.
+
+        Args:
+            path (str) - The directory to check
+            depth (int) - The maximum depth for recursive search
+            env (Environment) - A plugin collection. If not set, the default one
+                is used
```

Comment:
The last sentence is unsatisfactory, IMO. What is "the default one"? You could describe it as "only built-in plugins are used", or, more indirectly, as "a new `Environment` instance is created".

Reply:
Fixed